modules/python/test/test_mat.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+from __future__ import print_function
+
+import numpy as np
+import cv2 as cv
+
+import os
+import sys
+import unittest
+
+from tests_common import NewOpenCVTests
+
+try:
+    if sys.version_info[:2] < (3, 0):
+        raise unittest.SkipTest('Python 2.x is not supported')
+
+
+    class MatTest(NewOpenCVTests):
```

Comment:
2 practical use cases to test:
1. Create a `Mat` wrapper for `ndarray` and try to use one of the NumPy `ufunc`s e.g.:
```python
data = np.arange(10)
mat_data = cv.Mat(data)
np.testing.assert_equal(2 * data, 2 * mat_data)
```
2. Should 2 `Mat` objects constructed with different value of `wrap_channels` property be treated as equals?
```python
data = np.ones((10, 10, 3))
mat_wrapped = cv.Mat(data, wrap_channels=True)
mat_simple = cv.Mat(data)
self.assertEqual(mat_wrapped, mat_simple) # ???
```

Reply:
@VadimLevin Thank you for the review!

I added `ufunc` test case.
Comparison is tricky for now (numpy `==` returns an array which can't be casted to boolean result without `.any`/`.all` calls), so I would like to postpone introducing of custom comparison overloads.